virtualhome/grails-app/domain/aaf/vhr/ManagedSubject.groovy

```diff
@@ -515,15 +515,20 @@ class ManagedSubject {
       checkedNumber = "+64$checkedNumber"
     }
 
-    checkedNumber = checkedNumber.replace(' ','')
+    // Silently remove anything that might be entered by users in a valid-looking phone number.
+    checkedNumber = checkedNumber.replaceAll("[ .\\-()]", '')
```

Comment:
This regular expression might be doing something different than what you intended: `\\-(` might translate to  range from `\` to `(`.

The syntax for `[]` is that a `-` should come right after the opening `[` to be unambigous (not mean  a range).

Reply:
Btw, the  Regexp implementation is  used from Java - so https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html